api/src/main/java/io/opentelemetry/api/common/ArrayBackedAttributesBuilder.java

```diff
@@ -90,8 +90,11 @@ public AttributesBuilder put(String key, boolean... value) {
   }
 
   @Override
-  public AttributesBuilder putAll(Attributes attributes) {
-    attributes.forEach(this::put);
+  @SuppressWarnings({"unchecked", "rawtypes"})
+  public AttributesBuilder putAll(ReadableAttributes attributes) {
+    // Attributes must iterate over their entries with matching types for key / value, so this
+    // downcast to the raw type is safe.
+    attributes.forEach((key, value) -> put((AttributeKey) key, value));
```

Comment:
Populating another attributes is basically the only time `<T>` helped - but we can manage that ourselves without sacrificing the ability to lambda